fraud/proof.go

```diff
@@ -80,3 +93,40 @@ func Unmarshal(proofType ProofType, msg []byte) (Proof, error) {
 	}
 	return unmarshaler(msg)
 }
+
+// MonitorProofs enables metrics to monitor fraud proofs.
+func MonitorProofs(proofType ProofType) func(datastore.Datastore) {
```

Comment:
Let's design this around pkg as well. Similarly how we now join topics for any registered proof type on the pkg. So that any registered proof is automatically reported.

Reply:
Why? We join topics for any register proofs as it requires pubsub instance that is a part of our ProofService and also we do not want to shift responsibility of creating a topic to a user as it's a core part of our subscription. From the other side, metrics is `nice to have` but not required. In terms of our node, why do we need to observe some other fraud proofs?